commit 30110469145f3958690a4b0370fb9a17ee3b8a21
Author: Jason King <jason.king@joyent.com>
Date:   2019-07-31T20:46:37+00:00 (10 weeks ago)
    
    OS-7893 Early network admin causes network/physical service to fatally exit on certain bad configurations
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/overlay/generic/lib/svc/method/net-physical b/overlay/generic/lib/svc/method/net-physical
index d4868c6d..9d3231df 100644
--- a/overlay/generic/lib/svc/method/net-physical
+++ b/overlay/generic/lib/svc/method/net-physical
@@ -499,6 +499,16 @@ if smf_is_globalzone; then
 
     [[ -n "$EARLY_ADMIN" ]] || plumb_admin
 
+    # If we performed early configuration of the admin network and
+    # the admin interface is on a link aggregation, the aggregation
+    # has already been configured.  Add it to the list of active aggrs
+    # so the checks in vnic_up[6] are aware of it.
+    if [[ -n "$EARLY_ADMIN" && "${SYSINFO_NIC_admin}" =~ aggr[0-9]+$ ]]; then
+        eval "macs=\${SYSINFO_Aggregation_${SYSINFO_NIC_admin}_MACs}"
+
+        add_active_aggr_links ${SYSINFO_NIC_admin} $macs
+    fi
+
     # Prefer the config file for admin nic values, but use
     # bootparams if present
     admin_ip=${CONFIG_admin_ip}
